### PR TITLE
tool_operate: Fix error code on bad URL

### LINKED ISSUE #10130: curl 7.87.0 ftp 1800 files, get 2 errors - ftp timeout, followed by (27) out of memory
### I did this
I am using git-ftp v1.6.0 to transfer files from my machine to my VPS.
I am trying to upload 1300 files and delete 500 files
I get the following:
Files start to upload
ftp timeout error
curl continues then fails again with timeout error
Continues a third time then fails with Error(27) Out of memory

### I expected the following
all files to upload and delete according to git-ftp transfer log

### curl/libcurl version
curl 7.87.0 (x86_64-apple-darwin21.6.0) libcurl/7.87.0 (SecureTransport) OpenSSL/1.1.1s zlib/1.2.11 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.4 libssh2/1.10.0 nghttp2/1.51.0 librtmp/2.3
Release-Date: 2022-12-21

[curl -V output]
curl 7.87.0 (x86_64-apple-darwin21.6.0) libcurl/7.87.0 (SecureTransport) OpenSSL/1.1.1s zlib/1.2.11 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.4 libssh2/1.10.0 nghttp2/1.51.0 librtmp/2.3
Release-Date: 2022-12-21
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd

### operating system
Darwin xxx.local 21.6.0 Darwin Kernel Version 21.6.0: Sun Nov  6 23:31:16 PST 2022; root:xnu-8020.240.14~1/RELEASE_X86_64 x86_64

### PR DESCRIPTION
**Pull request by dfandrich**
TODO: There are probably other cases where a CURLUcode should be
converted into a CURLcode.

Closes #10130

### CODE REVIEW

**Comment by bagder** [comment]
I could not spot any other places...

**Comment by dfandrich** [comment]
I found a couple of other places where the wrong error code might be returned. I've also added a test case to check the original problem. This should be ready now.

File: src/tool_operate.c [hunk 1]
```diff
@@ -1209,21 +1209,26 @@ static CURLcode single_transfer(struct GlobalConfig *global,
           CURLU *uh = curl_url();
           if(uh) {
             char *updated;
-            if(curl_url_set(uh, CURLUPART_URL, per->this_url,
-                            CURLU_GUESS_SCHEME)) {
-              result = CURLE_FAILED_INIT;
+            CURLUcode uerr;
```
Comment (bagder) [inline src/tool_operate.c:1212]: You can move the `updated` declaration down to the `else {` clause now... And maybe consider initing it to NULL just to silence the static code analyzer.

Comment (dfandrich) [reply src/tool_operate.c:1211]: Done.

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
Thanks, merged it for you!

**Pull request closed without merging**